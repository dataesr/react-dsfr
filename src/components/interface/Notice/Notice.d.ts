import * as React from 'react';

export type NoticeClassName = string | Object | any[];

export type NoticeChildren = React.ReactNode[] | React.ReactNode | string;

export interface NoticeProps {
  className?: NoticeClassName;
  children?: NoticeChildren;
  title: string;
  visible?: boolean;
  asLink?: React.ReactElement<any>;
}

declare const Notice: React.FC<NoticeProps>;

export default Notice;
