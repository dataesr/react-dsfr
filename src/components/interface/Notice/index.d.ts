import type Notice from './Notice';

export type { Notice };
