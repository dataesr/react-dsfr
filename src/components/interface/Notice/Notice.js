import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import dataAttributes from '../../../utils/data-attributes';

/*
* DSFR v1.5.0
*/
import '@gouvfr/dsfr/dist/component/link/link.css';

const Notice = ({ chlidren, className, ...remainingProps }) => {
  const _className = classNames('fr-notice fr-notice--info', className, {});

  return (
    <div className={_className} {...dataAttributes.getAll(remainingProps)}>
      <div className="fr-container">
        <div className="fr-notice__body">
          <p className="fr-notice__title">
            Titre du bandeau,
            <a href="#" rel="noopener" target="_blank">lien au fil du texte</a>
          </p>
        </div>
      </div>
    </div>
  );
};

Notice.defaultProps = {
  className: '',
  children: '',
};

Notice.propTypes = {
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array,
  ]),
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
    PropTypes.string,
  ]),
};
export default Notice;
