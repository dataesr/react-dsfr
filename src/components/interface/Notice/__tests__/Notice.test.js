import Enzyme, { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import React from 'react';
import Notice from '../Notice';

Enzyme.configure({ adapter: new Adapter() });

const initialProps = {
  title: 'TiTle',
  visible: true,
};

describe('<Notice />', () => {
  it('renders correctly', () => {
    const component = renderer
      .create(
        <Notice
          title="Title"
          visible={initialProps.visible}
        />,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should call onClose', () => {
    const mockClick = jest.fn();

    const notice = shallow(
      <Notice
        visible
        onClose={mockClick}
        title="Title"
      />,
    );

    notice.find('.fr-btn--close').simulate('click');
    expect(mockClick).toHaveBeenCalled();
  });
});
