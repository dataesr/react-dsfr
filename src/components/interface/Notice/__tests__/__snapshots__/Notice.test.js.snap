// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notice /> renders correctly 1`] = `
<div
  className="fr-notice fr-notice--info"
>
  <div
    className="fr-container"
  >
    <div
      className="fr-notice__body"
    >
      <p
        className="fr-notice__title"
      >
        Title
        . 
      </p>
      <button
        className="fr-btn--close fr-btn"
        onClick={null}
        title="Masquer le message"
        type="button"
      >
        Masquer le message
      </button>
    </div>
  </div>
</div>
`;
